x-pack/plugins/security_solution/public/cases/components/case_view/index.tsx

```diff
@@ -479,33 +489,104 @@ export const CaseComponent = React.memo<CaseProps>(
 );
 
 export const CaseView = React.memo(({ caseId, subCaseId, userCanCrud }: Props) => {
+  const {
+    cases: casesUi,
+    application: { navigateToApp },
+  } = useKibana().services;
+  const history = useHistory();
   const { data, isLoading, isError, fetchCase, updateCase } = useGetCase(caseId, subCaseId);
-  if (isError) {
-    return null;
-  }
+  const { formatUrl, search } = useFormatUrl(SecurityPageName.case);
+  const { formatUrl: detectionsFormatUrl, search: detectionsUrlSearch } = useFormatUrl(
+    SecurityPageName.detections
+  );
+
+  // TODO: better understand why one needs to be formatted and the other doesn't
+  const allCasesLink = getCaseUrl(search);
+  const formattedAllCasesLink = formatUrl(allCasesLink);
+  const backToAllCasesOnClick = useCallback(
+    (ev) => {
+      ev.preventDefault();
+      history.push(allCasesLink);
+    },
+    [allCasesLink, history]
+  );
+  const caseDetailsLink = formatUrl(getCaseDetailsUrl({ id: caseId }), { absolute: true });
+  const getCaseDetailHrefWithCommentID = (commentId: string) => {
+    return formatUrl(getCaseDetailsUrlWithCommentId({ id: caseId, commentId, subCaseId }), {
+      absolute: true,
+    });
+  };
+
+  const configureCasesHref = formatUrl(getConfigureCasesUrl());
+  const onConfigureCasesNavClick = useCallback(
+    (ev) => {
+      ev.preventDefault();
+      history.push(getConfigureCasesUrl(search));
+    },
+    [history, search]
+  );
+
+  const onDetectionsRuleDetailsClick = useCallback(
+    (ruleId: string | null | undefined) => {
+      navigateToApp(`${APP_ID}:${SecurityPageName.detections}`, {
+        path: getRuleDetailsUrl(ruleId ?? ''),
+      });
+    },
+    [navigateToApp]
+  );
+
+  const getDetectionsRuleDetailsHref = useCallback(
+    (ruleId) => {
+      return detectionsFormatUrl(getRuleDetailsUrl(ruleId ?? '', detectionsUrlSearch));
+    },
+    [detectionsFormatUrl, detectionsUrlSearch]
+  );
+
+  const { selectedPatterns: selectedAlertPatterns } = useSourcererScope(
+    SourcererScopeName.detections
+  );
 
-  if (isLoading) {
+  if (USE_RAC_CASES_UI) {
```

Comment:
can i convince you to do this in the `x-pack/plugins/security_solution/public/cases/pages/case_details.tsx`? Thats where i did configure and it was a bit easier since the `pages` files already have `history`. Anything that doesnt also go in a modal maybe we just put in the page directly??